luzer/luzer.c

```diff
@@ -324,6 +317,133 @@ load_custom_mutator_lib(void) {
 	return 0;
 }
 
+/**
+ * Tries to asses how much bytecode there are loaded.
+ *
+ * I looked into lua's introspection capabilities, could not find anything good.
+ * There is https://github.com/leegao/see.lua but I don't think it is a good idea
+ * to make a PR with code for 5 different versions of a library to someone else's lib.
+ * Their idea is simple - decode bytecode in runtime.
+ * There is also https://github.com/siffiejoe/lua-getsize but reasoning is the same.
+ * They take signatures for 'struct Prototype' with them. Having them allow to
+ * see what Lua interpreter thinks sizes are.
+ * So here we sit, in quite a pickle, yearning for a lua-native crossplatform solution.
+ *
+ * Basically, this is stupid and straigtforward - table tree walk from '_G'.
+ * '_G' is Lua's special table for global stuff.
+ * 'string.dump' works even in latest LuaJIT. Bytecode is not crossplatform but we don't
+ * need that.
+ * This will count everything in global scope and in proper packages due to 'package.loaded'.
+ * I found no way to access anything local without a reference to an activation record.
+ *
+ * It may be possible to find every stack somehow and walk every frame and do a 'getlocal'
+ * and 'getupvalue' on them. No 'getstack' from within Lua tho, so one will have to write
+ * that in C.
+ * With 'struct Prototype' locals would be a cakewalk.
+ *
+ * This also can be written in C, but I see no reason for it. It should run only once.
+ * And C implementation would require much more time.
```

Comment:
To be honestly, I don't like a current implementation. However, I don't know what would be better. Agree, that rewriting to Lua C will probably a waste of time and probably be less maintainable. Probably, we should put a Lua code to a separate file and embed it on build stage, see [^1] and [^2].

This Lua function could be loaded on initial stage like `luaL_set_custom_mutator` and exported in `luzer` module, see [^3].


1. https://github.com/ligurio/lua-c-api-tests/commit/719cf90f65865cf5f2fe7fdb306cd75e1aa2f5e2
2. https://github.com/ligurio/lua-c-api-tests/commit/ba029382a1e5f0473a105ddc5136c8a9794fc833#diff-7fc5f49402ccaaa71949368218a21f5ee991358185ac6b01531b662259f9d585
3. https://github.com/ligurio/luzer/blob/f37950549bca59330117cbb44943b1984ab98b2c/luzer/luzer.c#L429C27-L429C50